powerstore/resource_storagecontainer.go

```diff
@@ -154,7 +163,73 @@ func (r resourceStorageContainer) Read(ctx context.Context, req tfsdk.ReadResour
 
 // Update resource
 func (r resourceStorageContainer) Update(ctx context.Context, req tfsdk.UpdateResourceRequest, resp *tfsdk.UpdateResourceResponse) {
+	log.Printf("Started Update")
+
+	// Get plan values
+	var plan models.StorageContainer
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get current state
+	var state models.StorageContainer
+	diags = req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// a worksround
+	// currently a bug on powerstore, not accepting PATCH call for same values
+
+	storageContainerUpdate := &gopowerstore.StorageContainer{}
+
+	if plan.Name.Value != state.Name.Value {
+		storageContainerUpdate.Name = plan.Name.Value
```

Comment:
this code can be made modular by moving these to the helper file.

Reply:
but this is only one time used code specific to here, so won't fulfil the modular paradigm here
but if it really solve or a better approach, I will move its in its own method
should I move ?

Reply:
Better to move different method

Reply:
moving